### ISSUE TITLE
Bug: object-curly-newline `ImportDeclaration` `never` still allows line breaks

### ISSUE THREAD

**Issue by StephanBijzitter** [eslint/eslint#15765]
### Environment

Node version: v14.19.0
npm version: v6.14.16
Local ESLint version: v8.12.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22593

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
                'object-curly-newline': ['error', {
                    ImportDeclaration: 'never'
                }],
```
</details>

Reported as error (expected):
```js
import {
    parseRange,
    isRangeSatisfiable,
    createContentRange
} from '../../../helpers/headers';
```

### What did you expect to happen?

`eslint --fix` to produce the following code:
```
import {parseRange, isRangeSatisfiable, createContentRange} from '../../../helpers/headers';
```
> After manually performing this change, ESLint does not report an error (so it is valid according to the rule and to my expectation)

### What actually happened?

`eslint --fix` changed the original to the following (and no longer reports it as error):
```
import {parseRange,
    isRangeSatisfiable,
    createContentRange} from '../../../helpers/headers';
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @StephanBijzitter, thanks for the issue!

This works as expected, [object-curly-newline](https://eslint.org/docs/rules/object-curly-newline) only requires/disallows linebreaks right after `{` and right before `}`.

> This rule requires or disallows a line break between `{` and its following token, and between `}` and its preceding token

We don't have a core rule that would require/disallow linebreaks between imported names.
